src/Microsoft.DotNet.ProjectModel/WorkspaceContext.cs

```diff
@@ -128,31 +145,93 @@ public void Refresh()
             _needRefresh = false;
         }
 
-        public IReadOnlyList<ProjectContext> GetProjectContexts(string projectPath)
+        public IReadOnlyList<ProjectContext> GetProjectContexts(string projectPath) => (IReadOnlyList<ProjectContext>)GetProjectContextCollection(projectPath)?.ProjectContexts.AsReadOnly() ?? EmptyArray<ProjectContext>.Value;
+        public ProjectContext GetProjectContext(string projectPath, NuGetFramework framework) => GetProjectContext(projectPath, framework, EmptyArray<string>.Value);
+        public ProjectContext GetProjectContext(string projectPath, NuGetFramework framework, IEnumerable<string> runtimeIdentifier)
         {
-            return GetProjectContextCollection(projectPath).ProjectContexts;
+            var contexts = GetProjectContextCollection(projectPath);
+            if (contexts == null)
+            {
+                return null;
+            }
+
+            return contexts
+                .ProjectContexts
+                .FirstOrDefault(c => Equals(c.TargetFramework, framework) && RidsMatch(c.RuntimeIdentifier, runtimeIdentifier));
+        }
+
+        public ProjectContext GetRuntimeContext(ProjectContext context, IEnumerable<string> runtimeIdentifiers)
+        {
+            var contexts = GetProjectContextCollection(context.ProjectDirectory);
+            if (contexts == null)
+            {
+                return null;
+            }
+
+            var runtimeContext = runtimeIdentifiers
+                .Select(r => contexts.GetTarget(context.TargetFramework, r))
+                .FirstOrDefault(c => c != null);
```

Comment:
The combination of `Select` and `FirstOrDefault` clauses feels like a bit waste. Only the first element of the `Select` clause result got used. This can be done with a loop cheaper.